photutils/psf/photometry.py

```diff
@@ -0,0 +1,450 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Module which provides classes to perform PSF Photometry"""
+
+from __future__ import division
+import numpy as np
+from astropy.table import Table, vstack, hstack
+from astropy.modeling.fitting import LevMarLSQFitter
+from astropy.nddata.utils import overlap_slices
+from astropy.stats import gaussian_sigma_to_fwhm
+from ..psf import subtract_psf
+from ..aperture import CircularAperture, aperture_photometry
+
+
+__all__ = ['DAOPhotPSFPhotometry']
+
+
+class DAOPhotPSFPhotometry(object):
+    """
+    This class implements the DAOPHOT algorithm proposed by Stetson
+    (1987) to perform point spread function photometry in crowded fields,
+    which consists basically in applying the loop FIND, GROUP, NSTAR,
+    SUBTRACT, FIND until no more stars are detected or a given number of
+    iterations is reached.
+    """
+
+    def __init__(self, group, bkg, psf, fitshape, find=None,
+                 fitter=LevMarLSQFitter(), niters=3, aperture_radius=None):
+        """
+        Attributes
+        ----------
+        group : callable or instance of any `~photutils.psf.GroupStarsBase`
+            subclasses ``group`` should be able to decide whether a given star
```

Comment:
"subclasses" needs to go at the end of the previous line.  Unfortunately, the parameter/type line cannot be wrapped.  It's a case where PEP8 line-length can be ignored since there is no alternative (that I'm aware of).